### PR TITLE
Integration test config with HRRR

### LINKED ISSUE #386: Add CI Integration Test for training
We should have an integration test that uses a full config to train a credit model with the CLI. The integration should include (in the config):

- At least two data sources
- At least two preblocks (including concat)
- At least two postblocks (including reconstruct)

To avoid having to regrid (and store regridding weights in the repo), we can use multiple streamable HRRR datasets. We should use a small domain size, and only train a few batches. 

We could also then run a rollout Integration directly after with the same model.   

This is dependent on #374 being completed first.

### PR DESCRIPTION
**Pull request by danineamati**
### Description

With development of CREDIT Gen2, there is a need for integration tests across the whole pipeline. Ideally, we want this integration test to run as separate from NCAR HPC as possible such that it could eventually act as a GitHub actions. This PR provides such an integration test, though it does not predict, yet.

Datasets:
* Remote HRRR PRS Subset [Prognostic, Dynamic Forcing, and Static]
* Remote HRRR SUBH Subset (same subset) [Prognostic and Diagnostic 

Preblocks:
* Log Transform
* Sqrt Transform
* Concat

Postblock:
* Reconstruct

Trainer: ERA5-Gen2 (Should we change the trainer name?)

Model: wxformer (59x47 with appropriate padding to 64x64)

Predict & PBS from the original and not tested.

Please consider running the following:

```shell
python scripts/multi_source_integration_check.py
```

and

```shell
credit train -c config/gen_2/smoke/integration_test_hrrr.yml
```

### Related Issues and PRs

Closes #386 

### Checklist
- [x] I am familiar with the [contributing guidelines](https://miles-credit.readthedocs.io/en/latest/contrib.html).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date.

### CODE REVIEW

File: credit/trainers/base_trainer.py [hunk 1]
```diff
@@ -170,6 +170,7 @@ def __init__(self, model: torch.nn.Module, rank: int, conf: Dict[str, Any]):
             ema_decay = trainer_conf.get("ema_decay", 0.9999)
             self.ema: Optional[EMATracker] = EMATracker(self.model, decay=ema_decay)
             ema_path = os.path.join(self.save_loc, "checkpoint_ema.pt")
+            # if self.load_weights and os.path.exists(ema_path):
```
Comment (danineamati) [inline credit/trainers/base_trainer.py]: I think this might help avoid having to delete the directory (i.e., checkpoints) each time you change the variables.

**Comment by kevinyang-cky** [comment]
@danineamati

Ran `python scripts/multi_source_integration_check.py` successfully and got

image

But `credit train -c config/gen_2/smoke/integration_test_hrrr.yml` failed...see the error message.

image

**Comment by danineamati** [comment]
> @danineamati
> 
> Ran `python scripts/multi_source_integration_check.py` successfully and got
> 
> image
> But `credit train -c config/gen_2/smoke/integration_test_hrrr.yml` failed...see the error message.
> 
> image

That likely means the folder is not empty. The current fix is to delete the directory where the model is going. We could also change the last child directory in `save_loc`. The commented-out line in `base_trainer` (see comment above) is a longer-term fix.

**Comment by kevinyang-cky** [comment]
After deleting the existing `/glade/derecho/scratch/$USER/CREDIT_runs/starter_gen2/` directory, the training went fine🥳

File: credit/trainers/base_trainer.py (hunk 1 above)
Comment (danineamati) [reply credit/trainers/base_trainer.py]: @jsschreck This is the line. Commented out is the recommendation.

File: config/gen_2/smoke/integration_test_hrrr.yml [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    total_steps: 500000  # ~ batches/epoch × total epochs
+    min_lr: 1.0e-5
+
+  amp: False
+  grad_accum_every: 1
+  grad_max_norm: 'dynamic'
+
+  thread_workers: 4
+  valid_thread_workers: 4
+  prefetch_factor: 4
```
Comment (kevinyang-cky) [inline config/gen_2/smoke/integration_test_hrrr.yml]: (more of an experience sharing) I typically use a `prefetch_factor` <= 2 and bump up the number of thread_workers, which also keeps GPUs fed as well.

File: config/gen_2/smoke/integration_test_hrrr.yml [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+        prognostic:
+          vars_3D: ['T', 'U', 'V', 'Q']
+          vars_2D: ['sp'] #, 't2m']
+
+        # Diagnostic: model predicts these but does not feed them back as input
+        diagnostic: null
+          # vars_2D: [ 'd2m' ]
+
+        # Dynamic forcing: time-varying inputs (not predicted)
+        dynamic_forcing:
```
Comment (kevinyang-cky) [inline config/gen_2/smoke/integration_test_hrrr.yml:39]: Quick question: Is there solar radiation at TOA in HRRR that could be brought in as dynamic_forcing?

**Review by kevinyang-cky — APPROVED** [review]
I just have some general comments; otherwise, LGTM! Also, I really like the inline comments, which make the config very easy to understand. Once the solar dataset is added, I plan to create a GOES/MRMS config similar to this one.

File: credit/trainers/base_trainer.py (hunk 1 above)
Comment (danineamati) [reply credit/trainers/base_trainer.py]: Switching to the check loading weights.

File: config/gen_2/smoke/integration_test_hrrr.yml (hunk 2 above)
Comment (danineamati) [reply config/gen_2/smoke/integration_test_hrrr.yml]: Changed to two.

File: config/gen_2/smoke/integration_test_hrrr.yml (hunk 3 above)
Comment (danineamati) [reply config/gen_2/smoke/integration_test_hrrr.yml:39]: Unfortunately, no TOA solar radiation to my knowledge.

**Comment by danineamati** [comment]
@djgagne, This PR now includes the updates to example 2026.2 as well. Feel free to merge when you and Charlie see fit.

**Review by djgagne — APPROVED** [review]
Looks good to me and should minimally conflict with Charlie's PR. Going ahead and merging.
